index.d.ts

```diff
@@ -0,0 +1,209 @@
+declare module 'whatwg-fetch' {
+  import {Agent} from 'http'
+  import {URLSearchParams, URL} from 'url'
+
+  export interface AbortSignal {
+    aborted: boolean
+
+    addEventListener: (
+      type: 'abort',
+      listener: (this: AbortSignal, event: any) => any,
+      options?:
+        | boolean
+        | {
+            capture?: boolean
+            once?: boolean
+            passive?: boolean
+          }
+    ) => void
+
+    removeEventListener: (
+      type: 'abort',
+      listener: (this: AbortSignal, event: any) => any,
+      options?:
+        | boolean
+        | {
+            capture?: boolean
+          }
+    ) => void
+
+    dispatchEvent: (event: any) => boolean
+
+    onabort?: null | ((this: AbortSignal, event: any) => void)
+  }
+
+  export class Request extends Body {
+    constructor(input: RequestInfo, init?: RequestInit)
+    clone(): Request
+    context: RequestContext
+    headers: Headers
+    method: string
+    redirect: RequestRedirect
+    referrer: string
+    url: string
+
+    // node-fetch extensions to the whatwg/fetch spec
+    agent?: Agent | ((parsedUrl: URL) => Agent)
+    compress: boolean
+    counter: number
+    follow: number
+    hostname: string
+    port?: number
+    protocol: string
+    size: number
+    timeout: number
+  }
+
+  export interface RequestInit {
+    // whatwg/fetch standard options
+    body?: BodyInit
+    headers?: HeadersInit
+    method?: string
+    redirect?: RequestRedirect
+    signal?: AbortSignal | null
+
+    // node-fetch extensions
+    agent?: Agent | ((parsedUrl: URL) => Agent) // =null http.Agent instance, allows custom proxy, certificate etc.
+    compress?: boolean // =true support gzip/deflate content encoding. false to disable
+    follow?: number // =20 maximum redirect count. 0 to not follow redirect
+    size?: number // =0 maximum response body size in bytes. 0 to disable
+    timeout?: number // =0 req/res timeout in ms, it resets on redirect. 0 to disable (OS limit applies)
```

Comment:
It looks like this was copied from `node-fetch` definitions.  I don't believe this library has any of the `node-fetch` extensions so they should not be included.

Reply:
Same for all of the node-fetch extensions throughout.

Reply:
Changing it now :D